### PR TITLE
Rework riscv interrupts

### PR DESCRIPTION
**Pull request by anton-bondarev**

### CODE REVIEW

File: src/include/hal/clock.h [hunk 1]
```diff
@@ -32,7 +32,7 @@ extern void clock_init(void);
  */
 extern void clock_setup(useconds_t useconds);
 
-extern void clock_tick_handler(int irq_num, void *dev_id);
```
Comment (alexkalmuk) [inline src/include/hal/clock.h:35]: It's required for risc-v somehow?

Comment (alexkalmuk) [reply src/include/hal/clock.h:35]: I mean removing `irq_num`.

Comment (alexkalmuk) [reply src/include/hal/clock.h:35]: Ah, it's just unused. So it's a drawback of the previous timer fixes.

Comment (anton-bondarev) [reply src/include/hal/clock.h:35]: Yes, you're right.
It's saved only for compatible with irq_atach()

**Pull request merged**